src/synchronizeContacts.js

```diff
@@ -20,6 +21,14 @@ const haveRemoteFieldsChanged = (
   )
 }
 
+// When merging contacts, we use the normal lodash merge function, except for the array of cozy URLs where we want a regular override
+const mergeWithArrayOverride = (objValue, srcValue, key) => {
```

Comment:
nit: maybe rename to `mergeWithCozyOverride` ? I prefer to see the field name in the function name as the special case is only for this field